test/e2e/data/infrastructure-docker/v1beta1/main/bases/cluster-with-kcp.yaml

```diff
@@ -59,6 +59,7 @@ spec:
       extraMounts:
         - containerPath: "/var/run/docker.sock"
           hostPath: "/var/run/docker.sock"
+      preLoadImages: ${DOCKER_PRELOAD_IMAGES:-[]}
```

Comment:
Ok let's just make this more explicit where the func is called

```suggestion
		// In case the infrastructure-docker provider is installed, ensure to add the preload images variable to load the
		// controller images into the nodes.
		// NOTE: we are checking the bootstrap cluster and assuming the workload cluster will be on the same
		// infrastructure provider. Also, given that we use it to set a variable, then it is up to cluster templates to use it or not.
		hasDockerInfrastructureProvider := hasProvider(ctx, input.BootstrapClusterProxy.GetClient(), "infrastructure-docker")


```

Reply:
added a comment to all usages of `DOCKER_PRELOAD_IMAGES`